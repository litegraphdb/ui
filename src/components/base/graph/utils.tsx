import { NodeData } from '@/lib/graph/types';
import { defaultNodeColor, nodeLightColorByType, universalNodeColors } from './constant';

const maxLegends = 16;

export const getLegendsForNodes = (nodes: NodeData[]) => {
  const priorityLegends = new Map<string, { legend: string; color: string }>();
  const customLegends = new Map<string, { legend: string; color: string }>();

  // Separate legends into priority (nodeLightColorByType) and custom
  nodes.forEach((node) => {
    if (Object.keys(nodeLightColorByType).includes(node.type)) {
      priorityLegends.set(node.type, { legend: node.type, color: nodeLightColorByType[node.type] });
    } else {
      // Only add if not already present to avoid changing colors for same legend
      if (!customLegends.has(node.type)) {
        customLegends.set(node.type, {
          legend: node.type,
          color: universalNodeColors[customLegends.size % universalNodeColors.length],
        });
      }
    }
  });

  const priorityLegendsSize = priorityLegends.size;
  // If there are nodes with unknown types, add "Unknown" to priority legends
  const isMoreThanMaxLegends = customLegends.size + priorityLegendsSize > maxLegends;

  // Convert to arrays for easier manipulation
  const priorityArray = Array.from(priorityLegends.values());
  const customArray = Array.from(customLegends.values()).slice(
    0,
    maxLegends - (priorityArray.length + (isMoreThanMaxLegends ? 1 : 0))
  );

  // Combine priority legends first, then custom legends
  const legends = [
    ...priorityArray,
    ...customArray,
    ...(isMoreThanMaxLegends ? [{ legend: 'Unknown', color: defaultNodeColor }] : []),
  ];
  const mapLegends: Record<string, { legend: string; color: string }> = {};
  legends.forEach((legend) => {
    mapLegends[legend.legend] = legend;
  });
  return mapLegends;
};
