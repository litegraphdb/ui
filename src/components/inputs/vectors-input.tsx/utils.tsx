export const convertVectorsToAPIRecord = (
  vectors?: Array<{ Model: string; Dimensionality: number; Content: string; Vectors: number[] }>
) => {
  return (
    vectors?.map((vector: any) => ({
      ...vector,
      Dimensionality: Number(vector.Dimensionality),
      Vectors: vector.Vectors || [],
    })) || []
  );
};
