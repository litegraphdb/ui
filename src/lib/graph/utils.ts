import type { NodeData, Point } from './types';

export function isPointInNode(point: Point, node: NodeData, radius: number = 10): boolean {
  const dx = point.x - node.x;
  const dy = point.y - node.y;
  return dx * dx + dy * dy <= radius * radius;
}

export function getNodeAtPosition(nodes: NodeData[], point: Point): NodeData | null {
  return nodes.find((node) => isPointInNode(point, node)) || null;
}

export function transformToOptions<T extends { GUID: string; name?: string; Name?: string }>(
  data?: T[] | null,
  labelField: keyof T = 'name' // Field to use for label in options
) {
  return data
    ? data?.map((item) => ({
        value: item.GUID,
        label: (item[labelField] as string) || item.Name || item.GUID,
      }))
    : [];
}
