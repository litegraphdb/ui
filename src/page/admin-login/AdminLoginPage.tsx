'use client';
import React, { useEffect, useState } from 'react';
import { Form, Input } from 'antd';
import { localStorageKeys } from '@/constants/constant';
import toast from 'react-hot-toast';
import {
  setAccessKey,
  setEndpoint,
  useGetTenants,
  useValidateConnectivity,
} from '@/lib/sdk/litegraph.service';
import { useAdminCredentialsToLogin } from '@/hooks/authHooks';
import { useAppDispatch } from '@/lib/store/hooks';
import { storeTenant } from '@/lib/store/litegraph/actions';
import LitegraphButton from '@/components/base/button/Button';
import LoginLayout from '@/components/layout/LoginLayout';
import { useCurrentlyHostedDomainAsServerUrl } from '@/hooks/appHooks';

interface AdminLoginFormData {
  url: string;
  accessKey: string;
}

const AdminLoginPage = () => {
  const dispatch = useAppDispatch();
  const [form] = Form.useForm();
  const [isServerValid, setIsServerValid] = useState(false);
  const { getTenants, isLoading } = useGetTenants();
  const loginWithAdminCredentials = useAdminCredentialsToLogin();
  const { validateConnectivity, isLoading: isValidating } = useValidateConnectivity();
  const serverUrl = useCurrentlyHostedDomainAsServerUrl();

  const handleValidateServerUrl = async () => {
    const url = form.getFieldValue('url');
    if (!url) return;
    setEndpoint(url);
    const isValid = await validateConnectivity();
    setIsServerValid(!!isValid);
  };

  const handleSubmit = async (values: AdminLoginFormData) => {
    try {
      setAccessKey(values.accessKey);
      const data = await getTenants('Login failed. Please try again.');
      if (data) {
        localStorage.setItem(localStorageKeys.adminAccessKey, values.accessKey);
        localStorage.setItem(localStorageKeys.serverUrl, values.url);
        loginWithAdminCredentials(values.accessKey);
        dispatch(storeTenant(data[0]));
      }
    } catch (error) {
      toast.error('Login failed. Please try again.');
    }
  };
  useEffect(() => {
    if (!serverUrl) return;
    form.setFieldValue('url', serverUrl);
    handleValidateServerUrl();
  }, [serverUrl]);

  return (
    <LoginLayout isAdmin>
      <Form
        form={form}
        layout="vertical"
        onFinish={handleSubmit}
        autoComplete="off"
        onKeyPress={(e) => {
          if (e.key === 'Enter') {
            form.submit();
          }
        }}
      >
        <Form.Item
          label="LiteGraph Server URL"
          name="url"
          rules={[
            { required: true, message: 'Please enter the LiteGraph Server URL!' },
            {
              validator: (_, value) => {
                if (!value) return Promise.resolve();
                try {
                  const parsedUrl = new URL(value);
                  if (!['http:', 'https:'].includes(parsedUrl.protocol)) {
                    return Promise.reject('Only HTTP or HTTPS URLs are allowed!');
                  }
                  return Promise.resolve();
                } catch (err) {
                  return Promise.reject('Please enter a valid URL!');
                }
              },
            },
          ]}
        >
          <Input
            placeholder="https://your-litegraph-server.com"
            onBlur={handleValidateServerUrl}
            disabled={isValidating}
          />
        </Form.Item>

        <Form.Item
          label="Access Key"
          name="accessKey"
          rules={[{ required: true, message: 'Please input your access key!' }]}
        >
          <Input.Password
            key={isServerValid ? 'enabled' : 'disabled'}
            placeholder="Enter your access key"
            onPressEnter={() => form.submit()}
            disabled={!isServerValid}
            autoFocus
          />
        </Form.Item>

        <Form.Item>
          <LitegraphButton
            type="primary"
            htmlType="submit"
            className="w-100"
            loading={isLoading || isValidating}
            disabled={isLoading || isValidating}
          >
            Login
          </LitegraphButton>
        </Form.Item>
      </Form>
    </LoginLayout>
  );
};

export default AdminLoginPage;
