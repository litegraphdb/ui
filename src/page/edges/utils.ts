import { EdgeType } from '@/lib/store/edge/types';
import { NodeType } from '@/lib/store/node/types';

// Utility to find the name by GUID
export const getNodeNameByGUID = (guid: string, nodesList: NodeType[]) => {
  if (!guid) return 'N/A';
  const node = nodesList.find((n: NodeType) => n.GUID === guid);
  return node ? node.Name : guid;
};

export const transformEdgeDataForTable = (edgesList: EdgeType[], nodesList: NodeType[]) => {
  return (
    edgesList?.map((data: EdgeType) => {
      return {
        ...data,
        FromName: getNodeNameByGUID(data.From, nodesList),
        ToName: getNodeNameByGUID(data.To, nodesList),
        key: data.GUID,
      } as EdgeType;
    }) || []
  );
};
