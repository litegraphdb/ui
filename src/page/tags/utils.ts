import { NodeType } from '@/lib/store/node/types';

import { getNodeNameByGUID } from '../edges/utils';
import { EdgeType } from '@/lib/store/edge/types';
import { TagMetaData } from 'litegraphdb/dist/types/types';
import { TagType } from '@/lib/store/tag/types';

// Utility to find the name by GUID
export const getEdgeNameByGUID = (guid: string, edgesList: EdgeType[]) => {
  if (!guid) return 'N/A';
  const edge = edgesList.find((n: EdgeType) => n.GUID === guid);
  return edge ? edge.Name : guid;
};

export const transformTagsDataForTable = (
  tagsList: TagMetaData[],
  nodesList: NodeType[],
  edgesList: EdgeType[]
) => {
  return (
    tagsList?.map((data: TagMetaData) => {
      return {
        ...data,
        NodeName: getNodeNameByGUID(data.NodeGUID, nodesList),
        EdgeName: getEdgeNameByGUID(data.EdgeGUID, edgesList),
        key: data.GUID,
      } as TagType;
    }) || []
  );
};
