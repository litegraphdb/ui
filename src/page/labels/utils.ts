import { NodeType } from '@/types/types';
import { getNodeNameByGUID } from '../edges/utils';

import { getEdgeNameByGUID } from '../tags/utils';
import { LabelMetadata } from 'litegraphdb/dist/types/types';
import { EdgeType } from '@/types/types';
import { LabelMetadataForTable } from './types';

export const transformLabelsDataForTable = (
  labelsList: LabelMetadata[],
  nodesList: NodeType[],
  edgesList: EdgeType[]
) => {
  return (
    labelsList?.map((data: LabelMetadata) => {
      return {
        ...data,
        NodeName: getNodeNameByGUID(data.NodeGUID, nodesList),
        EdgeName: getEdgeNameByGUID(data.EdgeGUID, edgesList),
        key: data.GUID,
      } as LabelMetadataForTable;
    }) || []
  );
};

export const getNodeAndEdgeGUIDsByLabelList = (labelsList: LabelMetadata[]) => {
  const nodeGUIDs = labelsList.map((label) => label.NodeGUID);
  const edgeGUIDs = labelsList.map((label) => label.EdgeGUID);
  return {
    nodeGUIDs: Array.from(new Set(nodeGUIDs)),
    edgeGUIDs: Array.from(new Set(edgeGUIDs)),
  };
};
