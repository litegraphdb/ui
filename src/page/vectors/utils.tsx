import { NodeType } from '@/types/types';
import { getNodeNameByGUID } from '../edges/utils';
import { EdgeType } from '@/types/types';
import { getEdgeNameByGUID } from '../tags/utils';
import { VectorMetadata } from 'litegraphdb/dist/types/types';

type VectorDisplayData = {
  GUID: string;
  TenantGUID: string;
  GraphGUID: string;
  NodeGUID: string;
  EdgeGUID: string;
  Model: string;
  Dimensionality: number;
  Content: string;
  Vectors: number[];
  CreatedUtc: string;
  LastUpdateUtc: string;
  NodeName: string;
  EdgeName: string;
  key: string;
  vectorString: string;
  vectorCount: number;
};

export const transformVectorsDataForTable = (
  vectorsList: VectorMetadata[],
  nodesList: NodeType[],
  edgesList: EdgeType[] = []
): VectorDisplayData[] => {
  return (
    vectorsList?.map((data) => ({
      ...data,
      NodeName: getNodeNameByGUID(data.NodeGUID, nodesList),
      EdgeName: getEdgeNameByGUID(data.EdgeGUID, edgesList),
      key: data.GUID,
      vectorString: Array.isArray(data.Vectors) ? data.Vectors.join(', ') : '',
      vectorCount: Array.isArray(data.Vectors) ? data.Vectors.length : 0,
    })) || []
  );
};

export const getNodeAndEdgeGUIDsByVectorList = (vectorsList: VectorMetadata[]) => {
  const nodeGUIDs = vectorsList.map((vector) => vector.NodeGUID);
  const edgeGUIDs = vectorsList.map((vector) => vector.EdgeGUID);
  return {
    nodeGUIDs: Array.from(new Set(nodeGUIDs)),
    edgeGUIDs: Array.from(new Set(edgeGUIDs)),
  };
};
