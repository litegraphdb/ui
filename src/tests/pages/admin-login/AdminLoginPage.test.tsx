import '@testing-library/jest-dom';
import React from 'react';
import { screen, fireEvent, waitFor } from '@testing-library/react';
import { createMockInitialState } from '../../store/mockStore';
import { mockTenantData } from '../mockData';
import AdminLoginPage from '@/app/login/admin/page';
import { renderWithRedux } from '@/tests/store/utils';
import { getServer } from '@/tests/server';
import { commonHandlers } from '@/tests/handler';
import { handlers } from './handler';

jest.mock('@/lib/sdk/litegraph.service', () => ({
  setAccessKey: jest.fn(),
  setEndpoint: jest.fn(),
  useGetTenants: jest.fn(() => ({
    getTenants: jest.fn(),
    isLoading: false,
  })),
  useValidateConnectivity: jest.fn(() => ({
    validateConnectivity: jest.fn(),
    isLoading: false,
  })),
}));

const server = getServer([...commonHandlers, ...handlers]);

describe('AdminLoginPage', () => {
  beforeAll(() => {
    server.listen();
  });

  afterEach(() => server.resetHandlers());
  afterAll(() => server.close());

  it('should render the AdminLoginPage with form fields', async () => {
    const initialState = createMockInitialState();
    const { container } = renderWithRedux(<AdminLoginPage />, initialState, undefined, true);

    expect(screen.getByLabelText('LiteGraph Server URL')).toBeVisible();
    expect(screen.getByLabelText('Access Key')).toBeVisible();
    expect(screen.getByRole('button', { name: /login/i })).toBeVisible();
    expect(screen.getByPlaceholderText('https://your-litegraph-server.com')).toBeVisible();
    expect(screen.getByPlaceholderText('Enter your access key')).toBeVisible();
    expect(container).toMatchSnapshot('initial table state');
  });

  it('should display validation errors for required fields', async () => {
    const initialState = createMockInitialState();
    renderWithRedux(<AdminLoginPage />, initialState, undefined, true);

    const submitButton = screen.getByRole('button', { name: /login/i });
    fireEvent.click(submitButton);

    await waitFor(() => {
      expect(screen.getByText('Please enter the LiteGraph Server URL!')).toBeVisible();
      expect(screen.getByText('Please input your access key!')).toBeVisible();
    });
  });

  it('should validate URL format and show error for invalid URL', async () => {
    const initialState = createMockInitialState();
    renderWithRedux(<AdminLoginPage />, initialState, undefined, true);

    const urlInput = screen.getByPlaceholderText('https://your-litegraph-server.com');
    fireEvent.change(urlInput, { target: { value: 'invalid-url' } });

    const submitButton = screen.getByRole('button', { name: /login/i });
    fireEvent.click(submitButton);

    await waitFor(() => {
      expect(screen.getByText('Please enter a valid URL!')).toBeVisible();
    });
  });

  it('should reject non-HTTP/HTTPS URLs', async () => {
    const initialState = createMockInitialState();
    renderWithRedux(<AdminLoginPage />, initialState, undefined, true);

    const urlInput = screen.getByPlaceholderText('https://your-litegraph-server.com');
    fireEvent.change(urlInput, { target: { value: 'ftp://example.com' } });

    const submitButton = screen.getByRole('button', { name: /login/i });
    fireEvent.click(submitButton);

    await waitFor(() => {
      expect(screen.getByText('Only HTTP or HTTPS URLs are allowed!')).toBeVisible();
    });
  });

  it('should validate server URL on blur', async () => {
    const mockValidateConnectivity = jest.fn().mockResolvedValue(true);
    const mockSetEndpoint = jest.fn();

    require('@/lib/sdk/litegraph.service').useValidateConnectivity.mockReturnValue({
      validateConnectivity: mockValidateConnectivity,
      isLoading: false,
    });
    require('@/lib/sdk/litegraph.service').setEndpoint = mockSetEndpoint;

    const initialState = createMockInitialState();
    renderWithRedux(<AdminLoginPage />, initialState, undefined, true);

    const urlInput = screen.getByPlaceholderText('https://your-litegraph-server.com');
    fireEvent.change(urlInput, { target: { value: 'https://example.com' } });
    fireEvent.blur(urlInput);

    await waitFor(() => {
      expect(mockSetEndpoint).toHaveBeenCalledWith('https://example.com');
      expect(mockValidateConnectivity).toHaveBeenCalled();
    });
  });

  it('should submit form when Enter key is pressed', async () => {
    const mockGetTenants = jest.fn().mockResolvedValue(mockTenantData);

    require('@/lib/sdk/litegraph.service').useGetTenants.mockReturnValue({
      getTenants: mockGetTenants,
      isLoading: false,
    });

    const initialState = createMockInitialState();
    renderWithRedux(<AdminLoginPage />, initialState, undefined, true);

    const urlInput = screen.getByPlaceholderText('https://your-litegraph-server.com');
    const accessKeyInput = screen.getByPlaceholderText('Enter your access key');

    fireEvent.change(urlInput, { target: { value: 'https://example.com' } });
    fireEvent.change(accessKeyInput, { target: { value: 'test-access-key' } });

    fireEvent.keyPress(accessKeyInput, { key: 'Enter', charCode: 13 });

    await waitFor(() => {
      expect(mockGetTenants).toHaveBeenCalled();
    });
  });

  it('should show loading state during server validation', async () => {
    require('@/lib/sdk/litegraph.service').useValidateConnectivity.mockReturnValue({
      validateConnectivity: jest.fn(),
      isLoading: true,
    });

    const initialState = createMockInitialState();
    renderWithRedux(<AdminLoginPage />, initialState);

    const urlInput = screen.getByPlaceholderText('https://your-litegraph-server.com');
    const submitButton = screen.getByRole('button', { name: /login/i });

    expect(urlInput).toBeDisabled();
    expect(submitButton).toBeDisabled();
  });

  it('should show loading state during login', async () => {
    require('@/lib/sdk/litegraph.service').useGetTenants.mockReturnValue({
      getTenants: jest.fn(),
      isLoading: true,
    });

    const initialState = createMockInitialState();
    renderWithRedux(<AdminLoginPage />, initialState, undefined, true);

    const submitButton = screen.getByRole('button', { name: /login/i });
    expect(submitButton).toBeDisabled();
    expect(submitButton).toHaveClass('ant-btn-loading');
  });

  it('should disable access key field when server is not validated', async () => {
    const initialState = createMockInitialState();
    renderWithRedux(<AdminLoginPage />, initialState);

    const accessKeyInput = screen.getByPlaceholderText('Enter your access key');
    expect(accessKeyInput).toBeDisabled();
  });

  it('should handle Enter key press in access key field', async () => {
    const mockGetTenants = jest.fn().mockResolvedValue(mockTenantData);

    require('@/lib/sdk/litegraph.service').useGetTenants.mockReturnValue({
      getTenants: mockGetTenants,
      isLoading: false,
    });

    const initialState = createMockInitialState();
    renderWithRedux(<AdminLoginPage />, initialState);

    const urlInput = screen.getByPlaceholderText('https://your-litegraph-server.com');
    const accessKeyInput = screen.getByPlaceholderText('Enter your access key');

    fireEvent.change(urlInput, { target: { value: 'https://example.com' } });
    fireEvent.change(accessKeyInput, { target: { value: 'test-access-key' } });

    fireEvent.keyDown(accessKeyInput, { key: 'Enter', code: 'Enter' });

    await waitFor(() => {
      expect(mockGetTenants).toHaveBeenCalled();
    });
  });

  it('should handle form submission with both loading states disabled', async () => {
    const mockGetTenants = jest.fn().mockResolvedValue(mockTenantData);

    require('@/lib/sdk/litegraph.service').useGetTenants.mockReturnValue({
      getTenants: mockGetTenants,
      isLoading: false,
    });
    require('@/lib/sdk/litegraph.service').useValidateConnectivity.mockReturnValue({
      validateConnectivity: jest.fn(),
      isLoading: false,
    });

    const initialState = createMockInitialState();
    renderWithRedux(<AdminLoginPage />, initialState, undefined, true);

    const submitButton = screen.getByRole('button', { name: /login/i });
    expect(submitButton).not.toBeDisabled();
    expect(submitButton).not.toHaveClass('ant-btn-loading');
  });

  it('should not call validateConnectivity if URL field is empty on blur', async () => {
    const mockValidateConnectivity = jest.fn();

    require('@/lib/sdk/litegraph.service').useValidateConnectivity.mockReturnValue({
      validateConnectivity: mockValidateConnectivity,
      isLoading: false,
    });

    const initialState = createMockInitialState();
    renderWithRedux(<AdminLoginPage />, initialState, undefined, true);

    const urlInput = screen.getByPlaceholderText('https://your-litegraph-server.com');
    fireEvent.blur(urlInput);

    expect(mockValidateConnectivity).not.toHaveBeenCalled();
  });

  it('should handle successful form submission and dispatch tenant to store', async () => {
    const mockGetTenants = jest.fn().mockResolvedValue(mockTenantData);
    const mockDispatch = jest.fn();

    require('@/lib/sdk/litegraph.service').useGetTenants.mockReturnValue({
      getTenants: mockGetTenants,
      isLoading: false,
    });

    // Mock useAppDispatch
    const mockUseAppDispatch = jest.fn(() => mockDispatch);
    jest.doMock('@/lib/store/hooks', () => ({
      useAppDispatch: mockUseAppDispatch,
    }));

    const initialState = createMockInitialState();
    renderWithRedux(<AdminLoginPage />, initialState, undefined, true);

    const urlInput = screen.getByPlaceholderText('https://your-litegraph-server.com');
    const accessKeyInput = screen.getByPlaceholderText('Enter your access key');

    fireEvent.change(urlInput, { target: { value: 'https://example.com' } });
    fireEvent.change(accessKeyInput, { target: { value: 'test-access-key' } });

    const submitButton = screen.getByRole('button', { name: /login/i });
    fireEvent.click(submitButton);

    await waitFor(() => {
      expect(mockGetTenants).toHaveBeenCalled();
    });
  });
});
