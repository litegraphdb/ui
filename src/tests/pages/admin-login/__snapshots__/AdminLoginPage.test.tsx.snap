// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminLoginPage should render the AdminLoginPage with form fields: initial table state 1`] = `
<div>
  <div
    class="userLoginPage ant-flex css-dev-only-do-not-override-98ntnt ant-flex-align-stretch ant-flex-vertical"
    style="gap: 20px;"
  >
    <div
      class="userLoginPageHeader mb pb pt pr pl ant-flex css-dev-only-do-not-override-98ntnt ant-flex-align-center ant-flex-justify-space-between"
    >
      <img
        alt="Litegraph Logo"
        height="40"
        src="/favicon.png"
      />
      <div
        class="ant-flex css-dev-only-do-not-override-98ntnt ant-flex-align-center"
        style="gap: 10px;"
      >
        <svg
          color="black"
          fill="none"
          height="20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="cursor: pointer; transform: rotate(90deg);"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            id="circle-mask-1"
          >
            <rect
              fill="white"
              height="100%"
              width="100%"
              x="0"
              y="0"
            />
            <circle
              fill="black"
              r="9"
            />
          </mask>
          <circle
            cx="12"
            cy="12"
            fill="black"
            mask="url(#circle-mask-1)"
          />
          <g
            stroke="currentColor"
            style="opacity: 1;"
          >
            <line
              x1="12"
              x2="12"
              y1="1"
              y2="3"
            />
            <line
              x1="12"
              x2="12"
              y1="21"
              y2="23"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="4.22"
              y2="5.64"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="18.36"
              y2="19.78"
            />
            <line
              x1="1"
              x2="3"
              y1="12"
              y2="12"
            />
            <line
              x1="21"
              x2="23"
              y1="12"
              y2="12"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="19.78"
              y2="18.36"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="5.64"
              y2="4.22"
            />
          </g>
        </svg>
        <a
          href="/login"
        >
          <span
            class="ant-typography css-dev-only-do-not-override-98ntnt"
          >
            Login as User
          </span>
        </a>
      </div>
    </div>
    <div
      class="loginTitle"
    >
      <h1
        class="ant-typography headingCommonStyles css-dev-only-do-not-override-98ntnt"
        style="font-weight: 600; font-size: 22px;"
      >
        Admin Login
      </h1>
      <div
        class="ant-typography loginDescription css-dev-only-do-not-override-98ntnt"
      >
        Please enter your access key to login
      </div>
    </div>
    <div
      class="loginBox"
    >
      <form
        autocomplete="off"
        class="ant-form ant-form-vertical css-dev-only-do-not-override-98ntnt"
      >
        <div
          class="ant-form-item css-dev-only-do-not-override-98ntnt"
        >
          <div
            class="ant-row ant-form-item-row css-dev-only-do-not-override-98ntnt"
          >
            <div
              class="ant-col ant-form-item-label css-dev-only-do-not-override-98ntnt"
            >
              <label
                class="ant-form-item-required"
                for="url"
                title="LiteGraph Server URL"
              >
                LiteGraph Server URL
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control css-dev-only-do-not-override-98ntnt"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    aria-required="true"
                    class="ant-input css-dev-only-do-not-override-98ntnt ant-input-outlined"
                    id="url"
                    placeholder="https://your-litegraph-server.com"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-form-item css-dev-only-do-not-override-98ntnt"
        >
          <div
            class="ant-row ant-form-item-row css-dev-only-do-not-override-98ntnt"
          >
            <div
              class="ant-col ant-form-item-label css-dev-only-do-not-override-98ntnt"
            >
              <label
                class="ant-form-item-required"
                for="accessKey"
                title="Access Key"
              >
                Access Key
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control css-dev-only-do-not-override-98ntnt"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <span
                    class="ant-input-affix-wrapper ant-input-disabled ant-input-affix-wrapper-disabled css-dev-only-do-not-override-98ntnt ant-input-outlined ant-input-password"
                  >
                    <input
                      aria-required="true"
                      class="ant-input ant-input-disabled css-dev-only-do-not-override-98ntnt"
                      disabled=""
                      id="accessKey"
                      placeholder="Enter your access key"
                      type="password"
                      value=""
                    />
                    <span
                      class="ant-input-suffix"
                    >
                      <span
                        aria-label="eye-invisible"
                        class="anticon anticon-eye-invisible ant-input-password-icon"
                        role="img"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="eye-invisible"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                          />
                          <path
                            d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                          />
                        </svg>
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-form-item css-dev-only-do-not-override-98ntnt"
        >
          <div
            class="ant-row ant-form-item-row css-dev-only-do-not-override-98ntnt"
          >
            <div
              class="ant-col ant-form-item-control css-dev-only-do-not-override-98ntnt"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <button
                    class="ant-btn css-dev-only-do-not-override-98ntnt ant-btn-primary w-100"
                    type="submit"
                  >
                    <span>
                      Login
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
